### PR TITLE
Show function signature for Golang buffers.

### LINKED ISSUE #4368: improvement: go layer autocomplete must show function signature
Hi guys, I think than would be cool than the autocomplete in the go layer would show the signature (expected arguments and result for every function) inside the box or in other windows (similar to how ternjs works) 

in my emacs config this works and show the function signature but I don't know why the spacemacs layer is  different, I suppose than must be a configuration in the layer because scala with ensime works as expected.

thanks!!!...

### PR DESCRIPTION
**Pull request by carlosgaldino**
This fixes #4368.

### CODE REVIEW

File: layers/+lang/go/packages.el [hunk 1]
```diff
@@ -93,7 +93,11 @@
 
 (when (configuration-layer/layer-usedp 'auto-completion)
   (defun go/post-init-company ()
-    (spacemacs|add-company-hook go-mode))
+    (spacemacs|add-company-hook go-mode)
+    (add-hook 'go-mode-hook
+              (lambda ()
+                (setq-local company-go-show-annotation t)
```
Comment (fbergroth) [inline layers/+lang/go/packages.el]: `company-go-show-annotation` is part of the `company-go` package, it should be set there.

Comment (carlosgaldino) [inline layers/+lang/go/packages.el]: In their repo? Or on `go/init-company-go` in this same file?

Comment (fbergroth) [inline layers/+lang/go/packages.el]: Right, in `go/init-company-go` . Replace the `:init` section with
```el
(progn
  (setq company-go-show-annotation t)
  (push 'company-go company-backends-go-mode))
```

Comment (carlosgaldino) [inline layers/+lang/go/packages.el]: Changed. Thank you.

**Comment by fbergroth** [comment]
:+1: although `company-tooltip-align-annotations` can be discussed. I personally prefer having it set to `nil` for the golang backend.

**Comment by carlosgaldino** [comment]
@fbergroth Yeah, you're right. It is much better not aligned. I removed it.

**Comment by syl20bnr** [comment]
Thank you !
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**